plugins/janus_sip.c

```diff
@@ -2541,76 +2536,66 @@ void janus_sip_sofia_callback(nua_event_t event, int status, char const *phrase,
+		if(m->type == JANUS_SDP_AUDIO) {
+			JANUS_LOG(LOG_VERB, "       Audio: %"SCNu16"\n", m->port);
+			if(m->port) {
 				session->media.has_audio = 1;
-				session->media.remote_audio_rtp_port = m->m_port;
-				session->media.remote_audio_rtcp_port = m->m_port+1;	/* FIXME We're assuming RTCP is on the next port */
+				session->media.remote_audio_rtp_port = m->port;
+				session->media.remote_audio_rtcp_port = m->port+1;	/* FIXME We're assuming RTCP is on the next port */
 			}
-		} else if(m->m_type == sdp_media_video) {
-			JANUS_LOG(LOG_VERB, "       Video: %lu\n", m->m_port);
-			if(m->m_port) {
+		} else if(m->type == JANUS_SDP_VIDEO) {
+			JANUS_LOG(LOG_VERB, "       Video: %"SCNu16"\n", m->port);
+			if(m->port) {
 				session->media.has_video = 1;
-				session->media.remote_video_rtp_port = m->m_port;
-				session->media.remote_video_rtcp_port = m->m_port+1;	/* FIXME We're assuming RTCP is on the next port */
+				session->media.remote_video_rtp_port = m->port;
+				session->media.remote_video_rtcp_port = m->port+1;	/* FIXME We're assuming RTCP is on the next port */
 			}
 		} else {
 			JANUS_LOG(LOG_WARN, "       Unsupported media line (not audio/video)\n");
-			m = m->m_next;
+			temp = temp->next;
 			continue;
 		}
 		JANUS_LOG(LOG_VERB, "       Media connections:\n");
-		if(m->m_connections) {
-			sdp_connection_t *c = m->m_connections;
-			while(c) {
-				if(c->c_address) {
-					g_free(session->media.remote_ip);
-					session->media.remote_ip = g_strdup(c->c_address);
-					JANUS_LOG(LOG_VERB, "         [%s]\n", session->media.remote_ip);
-				}
-				c = c->c_next;
-			}
-		}
-		JANUS_LOG(LOG_VERB, "       Media RTP maps:\n");
-		sdp_rtpmap_t *r = m->m_rtpmaps;
-		while(r) {
-			JANUS_LOG(LOG_VERB, "         [%u] %s\n", r->rm_pt, r->rm_encoding);
-			r = r->rm_next;
+		if(m->c_addr) {
+			g_free(session->media.remote_ip);
+			session->media.remote_ip = g_strdup(m->c_addr);
+			JANUS_LOG(LOG_VERB, "         [%s]\n", session->media.remote_ip);
 		}
 		JANUS_LOG(LOG_VERB, "       Media attributes:\n");
-		sdp_attribute_t *a = m->m_attributes;
-		while(a) {
-			if(a->a_name) {
-				if(!strcasecmp(a->a_name, "rtpmap")) {
-					JANUS_LOG(LOG_VERB, "         RTP Map:     %s\n", a->a_value);
-				} else if(!strcasecmp(a->a_name, "crypto")) {
-					JANUS_LOG(LOG_VERB, "         Crypto:      %s\n", a->a_value);
-					if(m->m_type == sdp_media_audio || m->m_type == sdp_media_video) {
+		GList *tempA = m->attributes;
+		while(tempA) {
+			janus_sdp_attribute *a = (janus_sdp_attribute *)tempA->data;
+			if(a->name) {
+				if(!strcasecmp(a->name, "rtpmap")) {
+					JANUS_LOG(LOG_VERB, "         RTP Map:     %s\n", a->value);
+				} else if(!strcasecmp(a->name, "crypto")) {
+					JANUS_LOG(LOG_VERB, "         Crypto:      %s\n", a->value);
+					if(m->type == JANUS_SDP_AUDIO || m->type == JANUS_SDP_VIDEO) {
 						gint32 tag = 0;
 						int suite;
-						char crypto[40];
+						char crypto[41];
```

Comment:
maybe make this bigger just in care it ever changes? it's just a buffer on the stack...